src/main/java/bisq/core/payment/payload/F2FAccountPayload.java

```diff
@@ -0,0 +1,129 @@
+    // PROTO BUFFER
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    private F2FAccountPayload(String paymentMethodName,
+                              String id,
+                              String countryCode,
+                              String contact,
+                              String city,
+                              String extraInfo,
+                              long maxTradePeriod,
+                              @Nullable Map<String, String> excludeFromJsonDataMap) {
+        super(paymentMethodName,
+                id,
+                countryCode,
+                maxTradePeriod,
+                excludeFromJsonDataMap);
+        this.contact = contact;
+        this.city = city;
+        this.extraInfo = extraInfo;
+    }
+
+    @Override
+    public Message toProtoMessage() {
+        PB.F2FAccountPayload.Builder builder = PB.F2FAccountPayload.newBuilder()
+                .setContact(contact)
+                .setCity(city)
+                .setExtraInfo(extraInfo);
+        final PB.CountryBasedPaymentAccountPayload.Builder countryBasedPaymentAccountPayload = getPaymentAccountPayloadBuilder()
+                .getCountryBasedPaymentAccountPayloadBuilder()
+                .setF2FAccountPayload(builder);
+        return getPaymentAccountPayloadBuilder()
+                .setCountryBasedPaymentAccountPayload(countryBasedPaymentAccountPayload)
+                .build();
+    }
+
+    public static PaymentAccountPayload fromProto(PB.PaymentAccountPayload proto) {
+        PB.CountryBasedPaymentAccountPayload countryBasedPaymentAccountPayload = proto.getCountryBasedPaymentAccountPayload();
+        PB.F2FAccountPayload f2fAccountPayloadPB = countryBasedPaymentAccountPayload.getF2FAccountPayload();
+        return new F2FAccountPayload(proto.getPaymentMethodId(),
+                proto.getId(),
+                countryBasedPaymentAccountPayload.getCountryCode(),
+                f2fAccountPayloadPB.getContact(),
+                f2fAccountPayloadPB.getCity(),
+                f2fAccountPayloadPB.getExtraInfo(),
+                proto.getMaxTradePeriod(),
+                CollectionUtils.isEmpty(proto.getExcludeFromJsonDataMap()) ? null : new HashMap<>(proto.getExcludeFromJsonDataMap()));
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // API
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public String getPaymentDetails() {
+        return "Face to Face - Contact: " + contact + ", city: " + city + ", additional information: " + extraInfo;
+    }
+
+
+    @Override
+    public String getPaymentDetailsForTradePopup() {
+        // We don't show here more as the makers extra data are the relevant for the trade. City has to be anyway the
+        // same for maker and taker.
```

Comment:
While it's true that the city has to be the same it might just be nice to have that confirmation that you're in the right city in case multiple f2f have been set up.